$OpenBSD: patch-jdk_src_solaris_classes_sun_nio_ch_DefaultAsynchronousChannelProvider_java,v 1.2 2014/03/13 13:47:10 dcoppa Exp $

Fix DefaultAsynchronousChannelProvider.create so it does not crash
("java.lang.InternalError: platform not recognized"), but instead
creates an instance of BsdAsynchronousChannelProvider()

--- jdk/src/solaris/classes/sun/nio/ch/DefaultAsynchronousChannelProvider.java.orig	Wed Aug 29 01:15:24 2012
+++ jdk/src/solaris/classes/sun/nio/ch/DefaultAsynchronousChannelProvider.java	Wed Oct 30 14:26:01 2013
@@ -50,7 +50,7 @@ public class DefaultAsynchronousChannelProvider {
             return new SolarisAsynchronousChannelProvider();
         if (osname.equals("Linux"))
             return new LinuxAsynchronousChannelProvider();
-        if (osname.contains("OS X"))
+        if (osname.contains("OS X") || osname.equals("OpenBSD"))
             return new BsdAsynchronousChannelProvider();
         throw new InternalError("platform not recognized");
     }
